### PR TITLE
Github action to execute the changed jupyter notebooks

### PR DESCRIPTION
**Pull request by balajialg**
I would like to tweak the working version of GA to test for the notebooks that went through a change as part of a PR. I want the result of notebook execution when PR gets created and not when PR gets merged. I didn't want to mess with the working GA so created a new one

### CODE REVIEW

File: .github/workflows/test_changed_notebooks.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Get changed files
+      id: changed-files-git
+      run: |
+        cd ${GITHUB_WORKSPACE}
+        ls -a
+        git status || true
```
Comment (ryanlovett) [inline .github/workflows/test_changed_notebooks.yml:37]: git doesn't behave properly when the files are owned by a different user than the current one. I think that is why it kept failing. You might try running `chown -R root .` right before running `git status`.

Comment (balajialg) [reply .github/workflows/test_changed_notebooks.yml:37]: Fixed it. Thanks

**Comment by balajialg** [comment]
@ryanlovett I addressed your comment. I am using the latest tag to pull the last built Data 100 image (instead of specifying the exact tag which willl change regularly). Do you see any challenge with this approach? I see a few listed [here](https://medium.com/@mccode/the-misunderstood-docker-tag-latest-af3babfd6375). Let me know what you think

**Comment by ryanlovett** [comment]
@balajialg I don't think we add the `latest` tag to any of our images. I believe `latest` is a convention, and registries do not automatically apply that tag.

**Comment by balajialg** [comment]
@ryanlovett Got it! So, it won't fetch the latest image from the registry? I am reverting my commit then.

**Comment by ryanlovett** [comment]
@balajialg Right, if no image has been manually tagged and pushed, (with any tag, not just `latest`), it won't exist in the registry.

**Comment by balajialg** [comment]
@ryanlovett Got it, I read conflicting accounts about "latest" tag. Certain folks highlighted that it pulls the latest or last built image from registry. Hence, I clarified with you.

**Pull request merged**